src/InboundServer/handlers.js

```diff
@@ -600,6 +646,57 @@ const putTransfersByIdError = async (ctx) => {
     ctx.response.body = '';
 };
 
+/**
+ * Handles PUT /thirdPartyRequests/transactions/{ID} request. 
+ * This is response to a POST /thirdPartyRequests/transactions request
+ */
+const putThirdPartyReqTransactionsById = async (ctx) => {
+    if (ctx.state.conf.enableTestFeatures) {
+        // we are in test mode so cache the request
+        const req = {
+            headers: ctx.request.headers,
+            data: ctx.request.body
+        };
+        const res = await ctx.state.cache.set(`callback_${ctx.state.path.params.ID}`, req);
+        ctx.state.logger.log(`Caching callback: ${util.inspect(res)}`);
+    }
+
+    // publish an event onto the cache for subscribers to action
+    const thirdPartyTrxnChannel = ThirdpartyTrxnModelOut.notificationChannel(ctx.state.path.params.ID);
```

Comment:
can you refactor the code of getting the channel name and publishing into the more reusable pattern?

Reply:
Ok